src/main/java/com/samcomo/dbz/global/redis/aop/DistributedLockAop.java

```diff
@@ -0,0 +1,61 @@
+package com.samcomo.dbz.global.redis.aop;
+
+import com.samcomo.dbz.global.exception.ErrorCode;
+import com.samcomo.dbz.global.redis.LockType;
+import com.samcomo.dbz.global.redis.exception.RedissonException;
+import java.lang.reflect.Method;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.redisson.api.RLock;
+import org.redisson.api.RedissonClient;
+import org.springframework.stereotype.Component;
+
+@Aspect
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class DistributedLockAop {
+
+  private final RedissonClient redissonClient;
+  private final AopForTransaction aopForTransaction;
+
+  @Around("@annotation(com.samcomo.dbz.global.redis.aop.DistributedLock)")
+  public Object lock(ProceedingJoinPoint joinPoint) throws Throwable{
+
+    MethodSignature signature = (MethodSignature) joinPoint.getSignature();
+    Method method = signature.getMethod();
+    DistributedLock distributedLock = method.getAnnotation(DistributedLock.class);
+
+    String lockName = distributedLock.key();
+    LockType type = distributedLock.lockType();
+    RLock lock = redissonClient.getLock(lockName + ":" + type);
```

Comment:
```suggestion
    String lockKey = distributedLock.key() + ":" + distributedLock.lockType();
    RLock lock = redissonClient.getLock(lockKey);
```

가독성 생각했을 때 이 방법도 좋은 것 같습니다!

Reply:
저는 코드가 너무 길어질까봐 저렇게 변수로 빼서 사용했었는데 오히려 가독성을 헤칠 수 있겠네요...! 참고할게요!